### ISSUE TITLE
False negative when union type is used in an invariant context

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5534]
The following should generate an error, but it doesn't.

```python
def func(v1: list[str | None]):
    v2: list[str | int | None] = v1
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.318, which I just published. It will also be included in a future release of pylance.
